pkg/asset/kubeconfig/kubeconfig_test.go

```diff
@@ -0,0 +1,229 @@
+package kubeconfig
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/stretchr/testify/assert"
+)
+
+type fakeAsset int
+
+var _ asset.Asset = fakeAsset(0)
+
+func (f fakeAsset) Dependencies() []asset.Asset {
+	return nil
+}
+
+func (f fakeAsset) Generate(map[asset.Asset]*asset.State) (*asset.State, error) {
+	return nil, nil
+}
+
+func TestKubeconfigGenerate(t *testing.T) {
+	testDir, err := ioutil.TempDir(os.TempDir(), "kubeconfig_test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(testDir)
+
+	rootCA := fakeAsset(0)
+	adminCertKey := fakeAsset(1)
+	kubeletCertKey := fakeAsset(2)
+	installConfig := fakeAsset(3)
+
+	rootCAState := &asset.State{
+		Contents: []asset.Content{
+			{
+				Name: "root-ca.key",
+				Data: []byte("THIS IS ROOT CA KEY DATA"),
+			},
+			{
+				Name: "root-ca.crt",
+				Data: []byte("THIS IS ROOT CA CERT DATA"),
+			},
+		},
+	}
+
+	adminCertState := &asset.State{
+		Contents: []asset.Content{
+			{
+				Name: "admin.key",
```

Comment:
nit: we can now use the `tls` package's public variables for these.  Not worth holding up the merge, but something for followup work.

Reply:
I'm in favor of using strings in the tests, kind like a double check to make sure the consts are defined as expected. Although it can add some overhead if we ever change the consts, but I think it should be fine.

Reply:
Besides, the test itself doesn't depend on the filename, so it can be whatever we want.